network/network.go

```diff
@@ -9,6 +9,11 @@ import (
 
 var ErrStopped = errors.New("network stopped")
 
+type UnexpectedNodeStopMsg struct {
+	Name     string
```

Comment:
I suggest that `Name` should be changed, as it is confusing about what this `Name` is. If it is always a node name, then `NodeName` maybe.